examples/oidc/README.md

```diff
@@ -0,0 +1,24 @@
+# Secure a service with OIDC authentication in a non-OIDC sercured Kubernetes cluster
```

Comment:
Let's set a short and tidy header, i.e. `# OIDC authentication` and explain the rationale in a first paragraph. Suggestion:

> Kubernetes supports OIDC authentication natively, configured via API server flags. In this case, a request contains the OIDC ID token in its bearer header. The token is then used for token review by the Kubernetes API. This already works with the existing kube-rbac-proxy functionality.  The resulting metadata (user and/or group information) is then passed to the subject review Kubernetes API for authorization.

> If Kubernetes is not configured to use OIDC and changes to the API server are not possible (i.e. in 3rd party or restricted environments), kube-rbac-proxy can be configured to authenticate the request against OIDC itself. In this case, the token review functionality of Kubernetes is omitted. As above the resulting metadata is still passed to the subject review Kubernetes API for authorization.

/cc @brancz